diesel/src/pg/expression/helper_types.rs

```diff
@@ -308,6 +308,29 @@ pub type NotLikeBinary<Lhs, Rhs> = crate::dsl::NotLike<Lhs, Rhs>;
 #[deprecated(note = "Use `dsl::Concat` instead")]
 pub type ConcatArray<Lhs, Rhs> = crate::dsl::Concat<Lhs, Rhs>;
 
+/// Return type of [`array_to_string_with_null_string(arr, delim, null_str)`](super::functions::array_to_string_with_null_string())
+#[allow(non_camel_case_types)]
+#[cfg(feature = "postgres_backend")]
+pub type array_to_string_with_null_string<A, E, N> =
+    super::functions::array_to_string_with_null_string<
+        SqlTypeOf<A>, // The SQL type of the array
+        SqlTypeOf<E>, // The SQL type of the delimiter
+        SqlTypeOf<N>, // The SQL type of the null_string
+        A,
+        E,
+        N,
+    >;
+
+/// Return type of [`array_to_string(arr, delim)`](super::functions::array_to_string())
+#[allow(non_camel_case_types)]
+#[cfg(feature = "postgres_backend")]
+pub type array_to_string<A, E> = super::functions::array_to_string<
+    SqlTypeOf<A>, // The SQL type of the array
+    SqlTypeOf<E>, // The SQL type of the elements within the array
+    A,
+    E,
+>;
```

Comment:
```suggestion
pub type array_to_string<A, D> = super::functions::array_to_string<
    SqlTypeOf<A>, // The SQL type of the array
    A,
    D,
>;
```

We don't need to have the `SqlTypeOf<D>` here as we don't need to be generic over that argument type. The `SqlTypeOf` part is only required for arguments that are generic.